sdk/eventhubs/azure-messaging-eventhubs/src/consumer_client.cpp

```diff
@@ -100,36 +124,51 @@ namespace Azure { namespace Messaging { namespace EventHubs {
         m_consumerClientOptions.ApplicationID,
         m_consumerClientOptions.CppStandardVersion);
 
-    return Azure::Core::Amqp::_internal::Connection{
-        m_fullyQualifiedNamespace, m_credential, connectOptions};
+    auto connection{Azure::Core::Amqp::_internal::Connection{
+        m_fullyQualifiedNamespace, m_credential, connectOptions}};
+#if ENABLE_RUST_AMQP
+    connection.Open(context);
+#endif
+    return connection;
+    (void)context;
   }
 
-  void ConsumerClient::EnsureConnection(std::string const& partitionId)
+  void ConsumerClient::EnsureConnection(
+      std::string const& partitionId,
+      Azure::Core::Context const& context)
   {
     std::unique_lock<std::recursive_mutex> lock(m_sessionsLock);
     if (m_connections.find(partitionId) == m_connections.end())
     {
-      m_connections.emplace(partitionId, CreateConnection(partitionId));
+      m_connections.emplace(partitionId, CreateConnection(partitionId, context));
     }
   }
 
   Azure::Core::Amqp::_internal::Session ConsumerClient::CreateSession(
-      std::string const& partitionId) const
+      std::string const& partitionId,
+      Azure::Core::Context const& context) const
   {
     SessionOptions sessionOptions;
     sessionOptions.InitialIncomingWindowSize
         = static_cast<uint32_t>((std::numeric_limits<int32_t>::max)());
 
-    return m_connections.at(partitionId).CreateSession(sessionOptions);
+    auto session{m_connections.at(partitionId).CreateSession(sessionOptions)};
+#if ENABLE_RUST_AMQP
+    session.Begin(context);
```

Comment:
Is the C++ Event Hubs client effectively just a no-op if `ENABLE_RUST_AMQP` isn't set? I'm curious since I see this condition a lot in this and other PRs, but seems the code is effectively useless without.

Reply:
No, if ENABLE_RUST_AMQP isn't set, we use the uAMQP AMQP transport - that's the default.

uAMQP doesn't have explicit connection start/stop and session begin/end operations, instead those are implicit based on creating a message sender and receiver.

It's one of my major architectural issues with uAMQP - connection and session state are implicitly managed rather than explicitly managed.

What's worse is that connections and sessions ARE explicitly managed - if you are running an AMQP server. The protocol experience is asymmetric, which is just horrible.

Reply:
And my big work item for next year is to basically make the formal switch from uAMQP to the Rust implementation. At this point, I'm confident enough in the Rust implementation's reliability that I feel like it's ok to switch.